src/Microsoft.Diagnostics.EventFlow.Inputs.Trace/TraceInput.cs

```diff
@@ -200,6 +201,12 @@ private void SubmitEventData(string message, TraceEventType level, int? id = nul
                     eventPayload["RelatedActivityID"] = relatedActivityId;
                 }
 
+                var eventContext = ContextResolver.GetRequestContext<object>();
+                if (eventContext != null)
+                {
+                    eventPayload["EventContext"] = eventContext;
```

Comment:
Given what Vance said in the other conversation (about OnEventWritten being called synchronously) I think what you have is fine. That is, it is not possible that the context will change between the time EventSource event method is called and when OnEventWritten/EventSource input is called